### PR TITLE
improve multi-core performance of qupsample_bilinear2d

### PR DESCRIPTION
**Pull request by mingfeima**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #74999
* #68328
* #72575
* #71903
* #69667
* __->__ #69601
* #69600
* #69599
* #69598
* #69517

Differential Revision: [D33353156](https://our.internmc.facebook.com/intern/diff/D33353156)

### CODE REVIEW

**Comment by VitalyFedyunin** [comment]
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33353156).

**Comment by Xia-Weiwen** [comment]
Hi @jerryzh168 We have a few int8-related PRs targeting 1.11. And this is one of them. Could you prioritize to review these PRs? Thanks.

**Comment by mingfeima** [comment]
TODO: optimize contiguous memory format as well, some model might run UpSample prior to Conv2d.

**Comment by VitalyFedyunin** [comment]
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33353156).

**Comment by mingfeima** [comment]
Please follow the notes from #69600

**Review by dzdang — APPROVED** [review]

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33353156).

**Comment by dzdang** [comment]
@mingfeima  Can you please also rebase this stack after fixing the lint issue I mentioned above? Thanks

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 1]
```diff
@@ -2055,88 +2055,90 @@ void qupsample_bilinear2d_nhwc_kernel(
...
+    const auto rwidth = area_pixel_compute_scale<float>(
+        input_width, output_width, align_corners, scales_w);
+
+    auto input_q_zero_point = input.q_zero_point();
+    auto output_q_zero_point = output.q_zero_point();
+    at::parallel_for(0, nbatch * output_height * output_width, 0, [&](int64_t begin, int64_t end) {
+      int64_t b{0}, h2{0}, w2{0};
+      data_index_init(begin, b, nbatch, h2, output_height, w2, output_width);
+
+      for (const auto i : c10::irange(begin, end)) {
```
Comment (mingfeima) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:2073]: suppress unused variable warning.

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 2]
```diff
@@ -2055,88 +2055,90 @@ void qupsample_bilinear2d_nhwc_kernel(
...
+        (void)i; //Suppress unused variable warning
```
Comment (mingfeima) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:2074]: Suppress unused variable warning

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33353156).

**Pull request closed without merging**